hubtools/src/lib.rs

```diff
@@ -368,26 +555,20 @@ impl RawHubrisImage {
                 .enclosed_name()
                 .ok_or_else(|| Error::BadFileInZip(file.name().to_owned()))?
                 .to_owned();
-            let path = outpath.iter().collect::<Vec<_>>();
-            let new_data = if path.len() == 2
-                && path[0].to_str() == Some("img")
-                && path[1].to_str().unwrap().starts_with("final.")
-            {
-                Some(
-                    std::fs::read(temp_dir.path().join(path[1]))
-                        .map_err(Error::FileReadError)?,
-                )
-            } else {
-                None
-            };
-
-            out.start_file(outpath.to_slash().unwrap(), opts)?;
-            if let Some(data) = new_data {
-                out.write_all(&data).unwrap();
-            } else {
+
+            let path = outpath.to_slash().unwrap();
+            if !self.new_files.contains_key(&path) {
+                out.start_file(path, opts)?;
                 std::io::copy(&mut file, &mut out).unwrap();
             }
         }
+
+        // Write all of our new and modified files
+        for (f, d) in self.new_files.into_iter() {
+            out.start_file(f, opts)?;
+            out.write_all(&d).unwrap();
```

Comment:
They're not quite identical: we have to use `std::io::copy` above because the `ZipFile` implements `Read` but can't be converted to a `&[u8]`.

Here, we've got a `Vec<u8>`, so we can use `write_all`.  I think the existing code is more clear than
```rust
            std::io::copy(&mut d, &mut out).unwrap();
```
because that begs the question of whether you're mutating `d`.